src/main.rs

```diff
@@ -1,3 +1,41 @@
+#![feature(proc_macro_hygiene, decl_macro)]
+#[macro_use]
+extern crate rocket;
+
+// in v0.5.1-rc Json is part of rocket
+// https://stackoverflow.com/questions/68682054/how-to-return-json-as-a-response-in-rust-rocket-with-auto-field-deserialising
+// but we are still using 0.4.10 for now
+use rocket_contrib::json::Json;
+use serde::{Deserialize, Serialize};
+#[macro_use]
+extern crate serde_derive;
+
+#[derive(Responder, Debug)]
+pub enum BackendErr {
+    #[response(status = 403)]
+    Unauthorized(String),
+    #[response(status = 404)]
+    NotFound(String),
+}
+
+#[derive(Serialize, Clone, Debug)]
+pub struct Score {
+    pub id: i32,
+    pub score: i32,
+    pub player: String,
+}
+
+#[get("/list")]
+fn list() -> Result<Json<Vec<Score>>, BackendErr> {
```

Comment:
Is Result necessary here?

Reply:
I think it will be, we want to handle the error cases here right?